tests/test_sparse_kpca.py

```diff
@@ -0,0 +1,142 @@
+import unittest
+import warnings
+
+import numpy as np
+from sklearn import exceptions
+from sklearn.datasets import load_boston
+from sklearn.utils.validation import check_is_fitted
+
+from skcosmo.sparsified import SparseKPCA
+
+
+def _linear_kernel(X, Y=None):
+    if Y is None:
+        Y = X.copy()
+    return X @ Y.T
+
+
+class SparseKPCA_tests(unittest.TestCase):
+    def rel_error(self, A, B):
+        return np.linalg.norm(A - B) ** 2.0 / np.linalg.norm(A) ** 2.0
+
+    def setUp(self):
+        X, y = load_boston(return_X_y=True)
+        self.X = X
+        self.Y = y
+
+        self.error_tol = 1e-3
+        self.rounding = -int(round(np.log10(self.error_tol)))
+
+        self.kernels = ["linear", "poly", "rbf", "cosine", _linear_kernel]
+        self.n_active = 10
+
+    def check_kernel_types(self, kernel):
```

Comment:
> Wouldn't it be more in the spirit of the sklearn workflow to not add `selector` + `selector_args` but let this be constructed with a pipeline. Then `Sparsified` objects can be also be used with classes of the `sklearn.feature_selection` module.
> 
> EDIT: to be more clear `sklearn.feature_selection` is for feature selection, but you could use a simple transposer class to let it act as a selector of samples
> 
> Should we start with this PR to rename `Sparsified` -> `Nystroem`? Maybe even `NystroemKernel`.
> 
> Conceptually, I still dont understand the difference between this and using `sklearn.kernel_approximation.Nystroem` + PCA, except that it does not offer a selector. Maybe an example with a comparison would be good. I have taken the example from
> https://scikit-learn.org/stable/auto_examples/miscellaneous/plot_kernel_approximation.html#sphx-glr-auto-examples-miscellaneous-plot-kernel-approximation-py
> and adapted it a bit and you can see that SKPCA gives a higher accuracy (see below), but I am not sure why this is so.
> https://gist.github.com/agoscinski/03836cb83249ec5caade7b5af4fb18ec
> ![skpca](https://user-images.githubusercontent.com/2772557/105765516-da59a180-5f58-11eb-93c9-4e04616141e0.png)
> 
> Some code is not covered, it would be better to cover it, even if the code is trivial.

Hey @agoscinski,

Looking through some of your comments, I think it might be beneficial for you to re-read the sparse kernel methods section of @bhelfrecht's [paper](https://iopscience.iop.org/article/10.1088/2632-2153/aba9ef). This will  clarify the difference between sklearn nystroem and this nystroem, as well as explain the improvement of these methods. It'll also explain the notations of `pkt` and `pkx`.

Reply:
I actually think the sklearn Nystroem and our sparse formulation are effectively the same. I think the only difference, as @agoscinski notes, is that the current sklearn Nystroem just selects the features (representative samples) randomly, whereas we might typically use something like FPS (this might partially explain the performance difference). I find myself increasingly favoring creating a more flexible Nystroem class rather than individual sparse classes.

Reply:
I disagree -- I think that sklearn nystroem produces Knm, and Knm alone, as a new set of features for each training sample. This kernel is then intended to be used in any kernel model in place of a original features. So then K \approx \phi{Knm} \phi{Knm}.T, where \phi is the RKHS mapping. Am I wrong here? I've scoured through sklearn docs, but it doesn't seem to line up.

Reply:
I'm pretty sure `sklearn.kernel_approximation.Nystroem` returns the RKHS features, i.e., Phi_nm, not the kernel K_nm. The returned features are then meant to be used in linear methods, rather than kernel methods. The example at the bottom of the doc page for `Nystroem` seems to support this. The only difference between the sklearn construction of Phi_nm and the one we outline in the KPCovR paper is that sklearn tacks on an extra set of eigenvectors at the end. This changes the "raw" features, but this should be inconsequential, since the dot product results in the same kernel.

I think a more flexible, modified `Nystroem` would work quite nicely in a pipeline. Upon initialization, you could specify a selection type (random, FPS, CUR, etc.), or provide a list of indices (or perhaps even samples) that will serve as the representatives/basis. The "extra eigenvectors" thing is also a tweak we could make here. I quite like this idea, as it would (somewhat off-topic) also obviate the need for incremental kernel methods (#33, #35, #36), though we could construct _linear_ incremental PCovR and LR if we saw fit.

Reply:
@hurricane642 Just had a long chat with Ben -- I see the issue now, and his suggestion is good. Instead of directly developing SKPCA, SKRR, and SKPCovR, can you take `sklearn.kernel_approximation.Nystroem` and modify it to be a bit more flexible? Biggest change is that you should be able to pass a selector instance. to the constructor to use in place of lines 771 - 774 in the Nystroem code. @bhelfrecht two questions: Will these kernels be centered? This should work fairly trivially with SKPCovR, no?

Reply:
The features that come out of `sklearn.kernel_approximation.Nystroem` I don't think will be centered/scaled out-of-the-box. However, you can treat them just like any old `X`, so you should be able to center/scale them with a preprocessing class. It should give you the same end result as applying the sparse scaling/centering to a Knm kernel for use in a sparse kernel method. To get SKPCovR, you would just do linear feature space PCovR on the Nystroem-transformed features (modulo some optional extra regularization).